tests/system/cli_test.go

```diff
@@ -0,0 +1,162 @@
+//go:build system_test
+
+package system
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/tidwall/gjson"
+)
+
+func TestUnsafeResetAll(t *testing.T) {
+	// scenario:
+	// 	given a non-empty wasm dir exists in the node home
+	//  when `unsafe-reset-all` is executed
+	// 	then the dir and all files in it are removed
+
+	wasmDir := filepath.Join(workDir, sut.nodePath(0), "wasm")
+	require.NoError(t, os.MkdirAll(wasmDir, os.ModePerm))
+
+	_, err := os.CreateTemp(wasmDir, "testing")
+	require.NoError(t, err)
+
+	// when
+	sut.ForEachNodeExecAndWait(t, []string{"tendermint", "unsafe-reset-all"})
+
+	// then
+	sut.withEachNodeHome(func(i int, home string) {
+		if _, err := os.Stat(wasmDir); !os.IsNotExist(err) {
+			t.Fatal("expected wasm dir to be removed")
+		}
+	})
+}
+
+func TestVestingAccounts(t *testing.T) {
```

Comment:
I guess this was a some work to port it. I would have skipped this test as it is testing the genesis generation only, which is not part of wasmd.